src/test/java/io/reactivex/rxjava3/internal/operators/flowable/BlockingFlowableToIteratorTest.java

```diff
@@ -163,4 +167,28 @@ protected void subscribeActual(Subscriber<? super Integer> s) {
 
         it.next();
     }
+
+    @Test(expected = NoSuchElementException.class)
+    public void disposedIteratorHasNextReturns() {
+        Iterator<Integer> it = PublishProcessor.<Integer>create()
+                .blockingIterable().iterator();
+        ((Disposable)it).dispose();
```

Comment:
Is this really public API? Feels weird and awkward to use

Reply:
It is not the intended usage but force-casting "hacks" the Iterator would result in hangs.

There is no good way to have cancellation through Iterable/Iterator as it was never intended to work with resources. For-each doesn't work with iterators and add try-with-resources over Iterable requires tracking all Iterator spawns so that all of them can be cancelled if the for-each/iteration is broken out.